cpp/oneapi/dal/backend/transfer.hpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
```

Comment:
data_transfer.hpp instead of transfer.hpp?

Reply:
Don't think the word "data" adds more clarity here

Reply:
When I see the file `transfer.hpp` without opening it - I have no clue what it contains. So `data` gives me more clarity here, probably it is a personal thing.